### PR TITLE
Add typescript definition file

### PR DESCRIPTION
**Pull request by wcandillon**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    export module NotificationType {
+        const Remote = "remote_notification";
+        const NotificationResponse = "notification_response";
+        const WillPresent = "will_present_notification";
+        const Local = "local_notification";
+    }
+
+    interface Subscription {
```
Comment (ibratoev) [inline index.d.ts]: Please export that interface as well so that it can be used where needed.

**Comment by wcandillon** [comment]
@ibratoev makes sense 👍

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        const WillPresent = "will_present_notification";
+        const Local = "local_notification";
+    }
+
+    export interface Subscription {
+        remove(): void;
+    }
+
+    export class FCM {
+        static requestPermissions(): void;
```
Comment (DanielRamosAcosta) [inline index.d.ts]: Why `FCM#requestPermissions` returns `void` instead of `Boolean`?

/cc @evollu

Comment (evollu) [reply index.d.ts]: hmm, it should be Promise<void> actually.
check the source code, it doesn't return boolean. It rejects if failed to get permission

**Pull request merged**